optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/common/DroolsCondition.java

```diff
@@ -175,20 +175,29 @@ protected DroolsCondition(T ruleStructure) {
 
     protected <S extends Score<S>, H extends AbstractScoreHolder<S>> void impactScore(DroolsConstraint<?> constraint,
             Drools drools, H scoreHolder, int impact) {
+        if (impact == 0) { // No need to include and no need to undo later.
+            return;
+        }
```

Comment:
Same here: if a matchWeigth is zero, it probably shouldn't disable the indictment. For clarity: If the constraintWeight is zero, it should (and the entire constraint execution too, as it does in the rest of this PR).